src/ir_passes/bundle_elim.rs

```diff
@@ -0,0 +1,233 @@
+use std::collections::HashMap;
+
+use itertools::Itertools;
+
+use crate::ir::{
+    Access, Command, CompIdx, Component, Connect, Context, Ctx, DenseIndexInfo,
+    Expr, Foreign, InvIdx, Invoke, Liveness, MutCtx, Port, PortIdx, Range,
+    Time,
+};
+
+#[derive(Default)]
+pub struct BundleElim {
+    context: DenseIndexInfo<Component, HashMap<PortIdx, Vec<PortIdx>>>,
+}
+
+impl BundleElim {
+    fn port(&self, pidx: PortIdx, comp: &mut Component) -> Vec<PortIdx> {
+        let one = comp.add(Expr::Concrete(1));
+        let Port {
+            owner,
+            width,
+            live,
+            info,
+        } = comp.get(pidx).clone();
+
+        let Liveness { idx, len, range } = live;
+
+        let start = comp.get(range.start).clone();
+        let end = comp.get(range.end).clone();
+
+        let len = len.as_concrete(comp).unwrap();
```

Comment:
The length of a port should be a method on the `ir::PortIdx` object since it seems like an extremely common query.

Reply:
Not doing this here because the function requires every part of the deconstructed port, not just the length. If we find the helper function useful elsewhere we can probably add it there.

Reply:
That seems fine right? If you call Len on the port indices the port information should be in the cache anyways for the destructuring?